File: Leetcode/Java Codes/October/_04.java
import java.util.*;

class _04 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int height[] = new int[n];
        for(int i=0; i<n; i++){
            height[i] = sc.nextInt();
        }
        System.out.print(maxArea(height));
        sc.close();
    }

    static int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxA = 0;

        while(left <= right){
            int area = Math.min(height[right], height[left]) * (right-left);

            maxA = Math.max(maxA, area);

            if(height[left]<=height[right]) left++;
            else right--;
        }
        return maxA;
    }

}

File: Leetcode/Python/_04.py
class Solution:
    def helper(self, height : list) -> int:
        left=0
        right=len(height)-1
        maxA = 0

        while left<=right :
            area = min(height[left], height[right]) * (right-left)

            maxA = max(maxA, area)

            if height[left] <= height[right]: left+=1
            else: right-=1
        
        return maxA
    
def main():
    n = int(input())
    height = list(map(int, input().split()))
    sol = Solution()
    res = sol.helper(height)
    print(res)

if __name__ == "__main__":
    main()